### PR TITLE
add error handling and tests

### PR DESCRIPTION
**Pull request by gmrabian**
## Description

Progresses [MDCT-257](https://qmacbis.atlassian.net/browse/MDCT-257) - adds tests for app-api handlers. Additionally improves error handling for missing key.

### How to test

`cd services/app-api`
`yarn test`
`yarn coverage`

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

File: services/app-api/handlers/banners/delete.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { deleteBanner } from "./delete";
+
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:2]:
```suggestion
```

File: services/app-api/handlers/banners/delete.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { APIGatewayProxyEvent } from "aws-lambda";
+import { proxyEvent } from "../../utils/testing/proxyEvent";
+import { StatusCodes } from "../../utils/types/types";
+import {
+  NO_KEY_ERROR_MESSAGE,
+  UNAUTHORIZED_MESSAGE,
+} from "../../utils/constants/constants";
+
+jest.mock("../../utils/dynamo/dynamodb-lib", () => ({
+  __esModule: true,
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:12]: Is this line necessary for all the util mocks?

File: services/app-api/handlers/banners/delete.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      ...testEvent,
+      pathParameters: {},
+    };
+    const res = await deleteBanner(noKeyEvent, null);
+
+    expect(res.statusCode).toBe(500);
+    expect(res.body).toContain(NO_KEY_ERROR_MESSAGE);
+  });
+
+  test("Test bannerKey empty", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:67]:
```suggestion
  test("Test bannerKey empty throws 500 error", async () => {
```

File: services/app-api/handlers/banners/delete.test.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+  test("Test Successful Banner Deletion", async () => {
+    const res = await deleteBanner(testEvent, null);
+
+    expect(res.statusCode).toBe(StatusCodes.SUCCESS);
+    expect(res.body).toContain("fakeBannerTable");
+    expect(res.body).toContain("testKey");
+  });
+
+  test("Test bannerKey not provided", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:56]:
```suggestion
  test("Test bannerKey not provided throws 500 error", async () => {
```

File: services/app-api/handlers/banners/delete.test.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  headers: { "cognito-identity-id": "test" },
+  pathParameters: { bannerId: "testKey" },
+};
+
+describe("Test Delete Banner", () => {
+  beforeEach(() => {
+    process.env["BANNER_TABLE_NAME"] = "fakeBannerTable";
+  });
+
+  test("Test not authorized to delete banner", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:41]:
```suggestion
  test("Test not authorized to delete banner throws 403 error", async () => {
```

File: services/app-api/handlers/banners/get.test.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  flush: jest.fn(),
+}));
+
+const testEvent: APIGatewayProxyEvent = {
+  ...proxyEvent,
+  headers: { "cognito-identity-id": "test" },
+  pathParameters: { bannerId: "testKey" },
+};
+
+describe("Test Get Banner", () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/get.test.ts:43]:
```suggestion
describe("Test getBanner API method", () => {
```

File: services/app-api/handlers/banners/delete.test.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  flush: jest.fn(),
+}));
+
+const testEvent: APIGatewayProxyEvent = {
+  ...proxyEvent,
+  headers: { "cognito-identity-id": "test" },
+  pathParameters: { bannerId: "testKey" },
+};
+
+describe("Test Delete Banner", () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/delete.test.ts:36]:
```suggestion
describe("Test deleteBanner API method", () => {
```

File: services/app-api/handlers/banners/write.test.ts [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+}));
+
+const testEvent: APIGatewayProxyEvent = {
+  ...proxyEvent,
+  body: `{"type":"banner","title":"test banner","description":"test description","link":"test link","startDate":1000,"endDate":2000}`,
+  headers: { "cognito-identity-id": "test" },
+  pathParameters: { bannerId: "testKey" },
+};
+
+describe("Test Write Banner", () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/write.test.ts:37]:
```suggestion
describe("Test writeBanner API method", () => {
```

File: services/app-api/handlers/banners/write.test.ts [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+  beforeEach(() => {
+    process.env["BANNER_TABLE_NAME"] = "fakeBannerTable";
+  });
+
+  test("Test not authorized run of banner creation", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/write.test.ts:42]:
```suggestion
  test("Test unauthorized banner creation throws 403 error", async () => {
```

File: services/app-api/handlers/banners/write.test.ts [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+  test("Test Successful Run of Banner Creation", async () => {
+    const res = await writeBanner(testEvent, null);
+
+    expect(res.statusCode).toBe(StatusCodes.SUCCESS);
+    expect(res.body).toContain("test banner");
+    expect(res.body).toContain("test description");
+  });
+
+  test("Test bannerKey not provided", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/write.test.ts:57]:
```suggestion
  test("Test bannerKey not provided throws 500 error", async () => {
```

File: services/app-api/handlers/banners/write.test.ts [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+      ...testEvent,
+      pathParameters: {},
+    };
+    const res = await writeBanner(noKeyEvent, null);
+
+    expect(res.statusCode).toBe(500);
+    expect(res.body).toContain(NO_KEY_ERROR_MESSAGE);
+  });
+
+  test("Test bannerKey empty", async () => {
```
Comment (braxex) [inline services/app-api/handlers/banners/write.test.ts:68]:
```suggestion
  test("Test bannerKey empty throws 500 error", async () => {
```

File: services/app-api/package.json [hunk 12]
```diff
@@ -51,7 +51,10 @@
     ],
     "collectCoverageFrom": [
       "handlers/**/*.{ts,tsx}",
-      "libs/**/*.{ts,tsx}"
+      "utils/**/*.{ts,tsx}",
+      "!utils/constants/*",
+      "!utils/testing/*",
+      "!utils/types/*"
```
Comment (braxex) [inline services/app-api/package.json:57]: Beautiful

**Review by braxex — APPROVED** [review]
Looks amazing. I added some nits re: naming of things so we can use the `-t` flag while testing these because I'm lazy.

File: services/app-api/handlers/banners/delete.test.ts (hunk 2 above)
Comment (gmrabian) [reply services/app-api/handlers/banners/delete.test.ts:12]: Thank you for making me learn what this is. You are right. We don't need it.
ref: https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options

**Review by gmrabian — COMMENTED** [review]
remove spacing

**Review by BearHanded — APPROVED** [review]

**Review by TheDanMiller — APPROVED** [review]

**Pull request merged**